TestCompactionPickerScores flakes occasionally

TestCompactionPickerScores flakes occasionally, with below error:

```
--- FAIL: TestCompactionPickerScores (0.02s)
    datadriven.go:143: 
        testdata/compaction_picker_scores:142:
         000005
        output didn't match expected:
        @@ -1,6 +1,2 @@
        -num-entries: 5
        -num-deletions: 5
        -num-range-key-sets: 0
        -point-deletions-bytes-estimate: 164586
        -range-deletions-bytes-estimate: 0
        +(not found)
```

Above flaked on https://github.com/cockroachdb/pebble/pull/2694. It's unlikely that the cause was changes made in the PR, since the changes in the PR were in the metamorphic testing suite.

https://github.com/cockroachdb/pebble/actions/runs/5424471226/jobs/9863874389?pr=2694

@jbowens tried to repro:

```
21942 runs so far, 0 failures, over 10m55s
```

--- a/compaction_picker_test.go
+++ b/compaction_picker_test.go
@@ -1508,11 +1508,13 @@ func TestCompactionPickerScores(t *testing.T) {
 		case "maybe-compact":
 			buf.Reset()
 			d.mu.Lock()
+			d.opts.DisableAutomaticCompactions = false
 			d.maybeScheduleCompaction()
 			fmt.Fprintf(&buf, "%d compactions in progress:", d.mu.compact.compactingCount)
 			for c := range d.mu.compact.inProgress {
 				fmt.Fprintf(&buf, "\n%s", c)
 			}
+			d.opts.DisableAutomaticCompactions = true
 			d.mu.Unlock()
 			return buf.String()
 

--- a/testdata/compaction_picker_scores
+++ b/testdata/compaction_picker_scores
@@ -107,16 +107,16 @@ L5  	687 B  2.3
 L6  	321 K  -
 
 # Run a similar test as above, but this time the table containing the DELs is
-# ingested after the database is initialized, with table stats enabled and
-# automatic compactions enabled. When the ingested sstable's stats are loaded,
-# it should trigger an automatic compaction of the ingested sstable on account
-# of the high point-deletions-bytes-estimate value.
+# ingested after the database is initialized. When the ingested sstable's stats
+# are loaded and automatic compactions are re-enabled, it should trigger an
+# automatic compaction of the ingested sstable on account of the high
+# point-deletions-bytes-estimate value.
 #
 # This a regression test for an issue where the table stats collector wouldn't
 # attempt to schedule a compaction if a file only had compensation due to point
 # deletions and not range deletions.
 
-define lbase-max-bytes=65536  enable-table-stats=true auto-compactions=on
+define lbase-max-bytes=65536  enable-table-stats=true auto-compactions=off
 L6
   a.SET.1:<rand-bytes=65536>
   b.SET.1:<rand-bytes=65536>
@@ -147,6 +147,12 @@ num-deletions: 5
 num-range-key-sets: 0
 point-deletions-bytes-estimate: 164586
 range-deletions-bytes-estimate: 0
+
+maybe-compact
+----
+1 compactions in progress:
+5: 000005:a:#10,0-e:#10,0
+6: 000004:a#1,1-e#1,1
 
 scores
 ----

